src/components/Form/Toggle.tsx

```diff
@@ -76,97 +61,27 @@ export interface ToggleProps {
   id?: string;
 }
 
-interface ToggleState {
+export interface ToggleState {
   toggleState: boolean;
 }
 
-const componentOrElementWithProps = (
-  component: ReactElement<any> | FunctionComponent<any> | ComponentClass<any>,
-  props: object,
-) => {
-  if (!!component) {
-    if (React.isValidElement(component)) {
-      return React.cloneElement(component, props); // element
-    }
-    return React.createElement(
-      component as FunctionComponent<any> | ComponentClass<any>,
-      props,
-    ); // component
-  }
-  return;
-};
-
 export class Toggle extends Component<ToggleProps> {
-  state: ToggleState = {
-    toggleState: !!this.props.checked || !!this.props.defaultChecked || false,
-  };
-
-  componentWillReceiveProps(nextProps: ToggleProps) {
-    const { checked } = nextProps;
-    if (!!checked) {
-      this.setState({ toggleState: !!checked });
-    }
-  }
-
-  handleClick = () => {
-    const { checked, onClick } = this.props;
-    const { toggleState } = this.state;
-    if (checked === undefined) {
-      this.setState({ toggleState: !toggleState });
-    }
-    if (!!onClick) {
-      onClick({ toggleState: !toggleState });
-    }
-  };
-
   render() {
-    const {
-      disabled,
-      className,
-      children,
-      toggleInputProps,
-      toggleInputComponent,
-      'aria-label': ariaLabel,
-      'aria-labelledby': ariaLabelledBy,
-      checked: dissMissChecked,
-      defaultChecked: dissMissDefaultChecked,
-      onClick: dissMissOnClick,
-      id: propId,
-      ...passProps
-    } = this.props;
-    const { toggleState } = this.state;
+    const { withInput, id: propId, ...passProps } = this.props;
     const id = idGenerator(propId);
-    const newToggleInputProps = {
-      disabled,
-      'aria-label': ariaLabel,
-      'aria-labelledby': ariaLabelledBy,
-      checked: !!toggleState,
-      className: toggleInputClassName,
-      onChange: this.handleClick,
-      ...toggleInputProps,
+    const newToggleProps = {
       id,
+      ...passProps,
     };
 
     return (
-      <HtmlSpan
-        className={classnames(className, baseClassName, {
-          [toggleDisabledClassName]: !!disabled,
-        })}
-      >
-        {!!toggleInputComponent ? (
-          componentOrElementWithProps(toggleInputComponent, newToggleInputProps) // element
+      <React.Fragment>
+        {!!withInput ? (
```

Comment:
You can also just do `return !!withInput ? (...`

Reply:
I made it simpler and now returning with the unneeded React.Fragment. Committed.